Reflection_Engine/Compute/ParseEnum.cs

```diff
@@ -0,0 +1,46 @@
+﻿using System;
+using System.Collections.Generic;
+using System.ComponentModel;
+using System.Linq;
+using System.Reflection;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace BH.Engine.Reflection
+{
+    public static partial class Compute
+    {
+        /*******************************************/
+        /**** Public Methods                    ****/
+        /*******************************************/
+
+        public static T ParseEnum<T>(string value)
+        {
+            object result = ParseEnum(typeof(T), value);
+            if (result == null)
+                return default(T);
+            else
+                return (T)result;
+        }
+
+        /*******************************************/
+
+        public static object ParseEnum(Type enumType, string value)
+        {
+            if (Enum.IsDefined(enumType, value))
+                return Enum.Parse(enumType, value);
+            else
+            {
+                return Enum.GetValues(enumType).OfType<Enum>()
```

Comment:
As discussed in call. Probably good to add a new attribute for this going forwards, as Descriptions have been used for longer explanations, rather than a more human friendly name, in the code. Happy for that to happen in subsequent PR's though if we have an issue raised for it.

Reply:
Issue raised here: https://github.com/BHoM/BHoM/issues/1323